smallrye-reactive-messaging-kafka/src/main/java/io/smallrye/reactive/messaging/kafka/commit/KafkaThrottledLatestProcessedCommit.java

```diff
@@ -355,10 +366,14 @@ long clearLesserSequentiallyProcessedOffsetsAndReturnLargestOffset() {
                         break;
                     }
                     unProcessedTotal--;
-                    largestSequentialProcessedOffset = receivedOffsets.poll().getOffset();
+                    OffsetReceivedAt poll = receivedOffsets.poll();
+                    if (poll != null) {
+                        largestSequentialProcessedOffset = poll.getOffset();
+                    }
```

Comment:
This null check isn't strictly necessary since the code block is in a while received offset is not empty. But this is fine.

Reply:
Yes, some weird issues have been reported. So, better to be safe...